### PR TITLE
224 process detail: Basic page setup and datafetching

### PR DESCRIPTION
**Pull request by DutchBen**

### CODE REVIEW

File: packages/orchestrator-ui-components/src/types/types.ts [hunk 1]
```diff
@@ -165,6 +165,38 @@ export type Process = {
...
+    FAILED = 'failed',
+    PENDING = 'pending',
+    RUNNING = 'running',
+}
+
+export interface ProcessDetail {
+    processId: Process['processId'];
+    status: Process['status'];
+    createdBy: Process['createdBy'];
+    started: Process['started'];
```
Comment (ricardovdheijden) [inline packages/orchestrator-ui-components/src/types/types.ts:179]: You could replace this whole block with the Pick utility type

**Review by ricardovdheijden — APPROVED** [review]

File: packages/orchestrator-ui-components/src/graphqlQueries/processDetailQuery.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
+import { parse } from 'graphql';
+import { gql } from 'graphql-request';
+import { TypedDocumentNode } from '@graphql-typed-document-node/core';
+import { ProcessesDetailResult } from '../types';
+
+export const GET_PROCESS_DETAIL_GRAPHQL_QUERY: TypedDocumentNode<
+    ProcessesDetailResult,
+    { processId: string }
```
[the remark this answers is not in the document]
Comment (DutchBen) [reply packages/orchestrator-ui-components/src/graphqlQueries/processDetailQuery.ts:8]: The graphql query vars contain the pagination options. This is a one of that I felt can stay here until we get a lot fo these

File: packages/orchestrator-ui-components/src/pages/tasks/WFOTaskListPage.tsx [hunk 3]
```diff
@@ -29,6 +29,7 @@ export const WFOTaskListPage = () => {
             sortBy: {
                 // Todo: waiting for fix in backend -- currently the sortBy field id's are not matching with the returned data
                 // https://github.com/workfloworchestrator/orchestrator-ui/issues/91
+                // eslint-disable-next-line @typescript-eslint/ban-ts-comment
```
[the remark this answers is not in the document]
Comment (DutchBen) [reply packages/orchestrator-ui-components/src/pages/tasks/WFOTaskListPage.tsx:32]: Mwa. The TODO is aleady there and I get distracted everytime I see the warning

File: packages/orchestrator-ui-components/src/types/types.ts (hunk 1 above)
Comment (DutchBen) [reply packages/orchestrator-ui-components/src/types/types.ts:178]: This does the same and gives you a better overview of the type IMO

File: packages/orchestrator-ui-components/src/graphqlQueries/index.ts [hunk 4]
```diff
@@ -2,4 +2,8 @@ export * from './subscriptionsListQuery';
 export * from './productBlocksQuery';
 export * from './productsQuery';
 export * from './resourceTypesQuery';
+<<<<<<< HEAD
```
Comment (ricardovdheijden) [inline packages/orchestrator-ui-components/src/graphqlQueries/index.ts:5]: Think something went wrong here

File: packages/orchestrator-ui-components/src/pages/processes/WFOProcessDetailPage.tsx [hunk 5]
```diff
@@ -0,0 +1,216 @@
...
+    translationKey,
+    value = '',
+}: ProcessHeaderValueProps) => {
+    const t = useTranslations('processes.detail');
+    const { theme } = useOrchestratorTheme();
+    return (
+        <EuiFlexGroup
+            direction="column"
+            gutterSize="xs"
+            css={{
```
Comment (ricardovdheijden) [inline packages/orchestrator-ui-components/src/pages/processes/WFOProcessDetailPage.tsx:42]: Try to move the styling into a separate file

File: packages/orchestrator-ui-components/src/utils/getProductNamesFromProcess.ts [hunk 6]
```diff
@@ -0,0 +1,16 @@
+import { ProcessDetail } from '../types';
+
+export const getProductNamesFromProcess = (
+    process: ProcessDetail | undefined | Omit<ProcessDetail, 'status'>,
+): string => {
+    if (!process) return '';
+
+    const productNames: string[] =
+        process && process.subscriptions && process.subscriptions.page
```
Comment (ricardovdheijden) [inline packages/orchestrator-ui-components/src/utils/getProductNamesFromProcess.ts:9]: "process?.subscriptions?.page" could replace the whole line

File: packages/orchestrator-ui-components/src/graphqlQueries/index.ts (hunk 4 above)
Comment (DutchBen) [reply packages/orchestrator-ui-components/src/graphqlQueries/index.ts]: Yes! Fixed

File: packages/orchestrator-ui-components/src/pages/processes/WFOProcessDetailPage.tsx [hunk 7]
```diff
@@ -0,0 +1,256 @@
...
+        if (lastCompletedStep) {
+            const currentStepIndex = steps.indexOf(lastCompletedStep);
+            return steps[currentStepIndex + 1]
+                ? steps[currentStepIndex + 1].name
+                : steps[currentStepIndex].name;
+        }
+
+        return steps[0] ? steps[0].name : '';
+    };
+    console.log(process);
```
Comment (ricardovdheijden) [inline packages/orchestrator-ui-components/src/pages/processes/WFOProcessDetailPage.tsx:98]: Accidental console log?

File: packages/orchestrator-ui-components/src/pages/processes/WFOProcessDetailPage.tsx [hunk 8]
```diff
@@ -0,0 +1,256 @@
...
+                    textOverflow: 'ellipsis',
+                }}
+            >
+                {value}
+            </EuiText>
+        </EuiFlexGroup>
+    );
+};
+
+export const WFOProcessDetailPage = ({
```
Comment (ricardovdheijden) [inline packages/orchestrator-ui-components/src/pages/processes/WFOProcessDetailPage.tsx:65]: The component is rather big, maybe consider to split up in smaller sub components

File: packages/orchestrator-ui-components/src/pages/processes/WFOProcessDetailPage.tsx (hunk 7 above)
Comment (DutchBen) [reply packages/orchestrator-ui-components/src/pages/processes/WFOProcessDetailPage.tsx]: Fixed

**Pull request merged**